src/ILVerify/src/ILImporter.StackValue.cs

```diff
@@ -362,7 +362,7 @@ bool IsBinaryComparable(StackValue src, StackValue dst, ILOpcode op)
                         case StackValueKind.ObjRef:
                             return op == ILOpcode.beq || op == ILOpcode.beq_s ||
                                    op == ILOpcode.bne_un || op == ILOpcode.bne_un_s ||
-                                   op == ILOpcode.ceq;
+                                   op == ILOpcode.ceq || op == ILOpcode.cgt_un;
```

Comment:
I am not sure, but seems like PeVerify does not allow `bgt_un` for object references either. Here is the PeVerify code, that handles the comparison case. `compareForEuqality` is set to `false` for `bgt_un`, but this needs to be true for object references.

```cpp
    switch (opcode)
    {
    case ReaderBaseNS::CEE_CGT_UN: //                   CE:4)
        //            vstate, equality?, branch?
        compareForEquality = true;
        withBranch = false;
        break;
// ...
    case ReaderBaseNS::CEE_BGT_UN: //                   CG:b4)
        //            vstate, equality?, branch?
        compareForEquality = false;
        withBranch = true;
        break;

//...
    else if (v1.IsObjRef())
    {
        //must be compare for equality
        Verify(compareForEquality, MVER_E_STACK_UNEXPECTED); //"Cond not allowed on object types"

        VerifyAndReportFound(v2.IsObjRef(), v2, MVER_E_STACK_NOT_EQ);
    }
```